vm.go

```diff
@@ -701,6 +769,82 @@ func (vm *VirtualMachineSpec) String() string {
 	return string(jsonString)
 }
 
+func (vm *VirtualMachineSpec) modifySetting(ctx context.Context, doc *hcsschema.ModifySettingRequest) (err error) {
+	if doc.GuestRequest == nil || vm.gc == nil {
+		return vm.system.Modify(ctx, doc)
+	}
+
+	hostdoc := *doc
+	hostdoc.GuestRequest = nil
+	if doc.ResourcePath != "" && doc.RequestType == requesttype.Add {
+		err = vm.system.Modify(ctx, &hostdoc)
+		if err != nil {
+			return fmt.Errorf("adding VM resources: %s", err)
+		}
+		defer func() {
+			if err != nil {
+				hostdoc.RequestType = requesttype.Remove
+				rerr := vm.system.Modify(ctx, &hostdoc)
+				if rerr != nil {
+					log.G(ctx).WithError(err).Error("failed to roll back resource add")
+				}
+			}
+		}()
+	}
+	err = vm.gc.Modify(ctx, doc.GuestRequest)
+	if err != nil {
+		return fmt.Errorf("guest modify: %s", err)
+	}
+	if doc.ResourcePath != "" && doc.RequestType == requesttype.Remove {
```

Comment:
should we use a else or switch case? operationtype can be either Add or Remove...

Reply:
we only need to do this part for remove.  For remove, guest requests are served first and then host requests.

Reply:
not sure i follow. if this part (GuestRequest) needs to be done only for Remove, then shouldn't we return after handling Add?
if doc.ResourcePath != "" && doc.RequestType == requesttype.Add {
..
return nil
}

Reply:
if no guest request or external gcs is not used, invoke hcs to handle the request.

if with guest request and external gcs is used:
For Add, process host request and then guest request
For Remove, process guest request and then host request
For others, just handle quest request